### PR TITLE
feat: Migrate user transactions to fetchWithCatchTxError

### PR DESCRIPTION
**Pull request by memoyil**
Missing error toast when transaction fails on chain

To review: 
https://deploy-preview-2942--pancakeswap-dev.netlify.app/

### CODE REVIEW

**Comment by chef-jojo** [comment]
@memoyil do you we can migrate some of those error toast to [useCatchTxError](https://github.com/pancakeswap/pancake-frontend/blob/develop/src/hooks/useCatchTxError.tsx)? 
It will skip the error when user rejected and try to show the condition message from contract.

**Comment by memoyil** [comment]
> @memoyil do you think we can migrate some of those error toast to [useCatchTxError](https://github.com/pancakeswap/pancake-frontend/blob/develop/src/hooks/useCatchTxError.tsx)? It will skip the error when user rejected and try to show the condition message from contract.

On it

File: src/hooks/useApproveConfirmTransaction.tsx [hunk 1]
```diff
@@ -90,47 +87,34 @@ const useApproveConfirmTransaction = ({
...
-      toastSuccess(`${t('Transaction Submitted')}!`, <ToastDescriptionWithTx txHash={tx.hash} />)
+    const receipt = await fetchWithCatchTxError(async () => {
       dispatch({ type: 'approve_sending' })
-      const receipt = await tx.wait()
-      if (receipt.status) {
-        dispatch({ type: 'approve_receipt' })
-        onApproveSuccess({ state, receipt })
-      }
-    } catch (error) {
-      dispatch({ type: 'approve_error' })
```
Comment (ngphi) [inline src/hooks/useApproveConfirmTransaction.tsx:110]: if the request is failed, do we need to update this state ?

**Comment by chefilip** [comment]
Great work ! LGTM

File: src/views/Farms/components/DepositModal.tsx [hunk 2]
```diff
@@ -149,18 +146,9 @@ const DepositModal: React.FC<DepositModalProps> = ({
           }
           onClick={async () => {
             setPendingTx(true)
-            try {
-              await onConfirm(val)
```
Comment (chef-jojo) [inline src/views/Farms/components/DepositModal.tsx:153]: Why remove try catch?

Comment (chef-jojo) [reply src/views/Farms/components/DepositModal.tsx:153]: oops. onStake also use fetchWithCatchTxError.

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**